src/main/java/seedu/address/logic/commands/UnmarkAttendanceCommand.java

```diff
@@ -39,8 +40,12 @@ public CommandResult executeCommand(Model model) throws CommandException {
             throw new CommandException(Messages.MESSAGE_INVALID_PERSON_DISPLAYED_INDEX);
         }
         Person personToUnmark = lastShownList.get(targetIndex.getZeroBased());
-        model.unmarkAttendance(personToUnmark);
-        return new CommandResult(MESSAGE_SUCCESS);
+        if (personToUnmark instanceof Student) {
```

Comment:
Good check of object type to prevent accidentally trying to mark attendance of a Teacher